x/ibc/02-client/keeper/client.go

```diff
@@ -127,14 +133,15 @@ func (k Keeper) CheckMisbehaviourAndUpdateState(ctx sdk.Context, misbehaviour ex
 		return sdkerrors.Wrapf(types.ErrClientNotFound, "cannot check misbehaviour for client with ID %s", misbehaviour.GetClientID())
 	}
 
-	consensusState, found := k.GetClientConsensusStateLTE(ctx, misbehaviour.GetClientID(), uint64(misbehaviour.GetHeight()))
-	if !found {
-		return sdkerrors.Wrapf(types.ErrConsensusStateNotFound, "cannot check misbehaviour for client with ID %s", misbehaviour.GetClientID())
-	}
-
 	var err error
 	switch e := misbehaviour.(type) {
 	case ibctmtypes.Evidence:
+		// Get ConsensusState at TrustedHeight
+		consensusState, found := k.GetClientConsensusState(ctx, misbehaviour.GetClientID(), e.Header1.TrustedHeight)
```

Comment:
ditto. Also applies to solo-machine [evidence](https://github.com/cosmos/ics/tree/master/spec/ics-006-solo-machine-client#evidence)